TypeCobol/Compiler/Sql/CodeElements/Statements/LockTableStatement.cs

```diff
@@ -0,0 +1,43 @@
+﻿using JetBrains.Annotations;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Sql.Model;
+
+namespace TypeCobol.Compiler.Sql.CodeElements.Statements
+{
+
+    public enum LockMode
+    {
+        Shared,
+        Exclusive
+    }
+
+    public class LockTableStatement : SqlStatementElement
+    {
+        public TableViewCorrelationName Table { get; }
+
+        [CanBeNull] public SqlConstant PartitionId { get; }
+
+        public SyntaxProperty<LockMode> Mode { get; }
+
+        public SyntaxProperty<bool> IsPartitionInteger { get; }
```

Comment:
I don't understand this property. The absence of partition id is already visible when `PartitionId` is null. Compared to `IsImmediate` in `TruncateStatement` for example, this is different. In TRUNCATE statement, the keyword IMMEDIATE changes the behavior of the statement so we capture this info using a `SyntaxProperty<bool>`. Here the relevant information is the partition identifier (including its absence) and this is fully captured through the `PartitionId` property.
Please remove the `IsPartitionInteger` property and adapt code accordingly.